core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -528,6 +529,10 @@ object KafkaConfig {
     "If log.message.timestamp.type=CreateTime, the message will be rejected if the difference in timestamps exceeds " +
     "this specified threshold. This configuration is ignored if log.message.timestamp.type=LogAppendTime."
 
+  val LogDirFailureTimeoutMsDoc = "If the broker is unable to successfully communicate to the controller that some log " +
+    "directory has failed for longer than this time, and there's at least one partition with leadership on that directory, " +
```

Comment:
> and there's at least one partition with leadership

We aren't checking for this condition. We can either a) implement it; or b) keep it simple and drop this out of the configuration description.

Reply:
I'll do another round with this, there might be a way in `BrokerServer` to extract this information using the combination of `MetadataCache`, `ReplicaManager` and `LogManager`. I'll update you tomorrow about my findings.